### PR TITLE
Add statestore terraform IaC

### PR DESCRIPTION
**Pull request by sachinkundu**
# PR for issue 5292 -> 5480, 5481

## What is being addressed

Add state store to TRE templates
Task: Module integrated with TRE main IaC template
Task IaC template(s) created

## How is this addressed

Add TF for state store cosmosDB
Tables will created later.

Networking for enabling connection to AppServices Management API will be done later.

### CODE REVIEW

**Review by marrobi — CHANGES_REQUESTED** [review]
Can you review the two points. Thanks.

File: templates/core/terraform/main.tf [hunk 1]
```diff
@@ -107,3 +107,10 @@ module "acr" {
   core_vnet             = module.network.core
   shared_subnet         = module.network.shared
 }
+
+module "state-store" {
+  source = "./state-store"
+  name = "tre-${var.resource_name_prefix}"
+  location = var.location
+  resource_group_name = azurerm_resource_group.core.name
+}
```
Comment (deniscep) [inline templates/core/terraform/main.tf:116]:
```suggestion
module "state-store" {
  source                = "./state-store"
  resource_name_prefix  = var.resource_name_prefix
  environment           = var.environment
  tre_id                = local.tre_id
  location              = var.location
  resource_group_name   = azurerm_resource_group.core.name
}
```
Suggest changing to the same pattern as the rest of the modules; also solves the naming convention that @marrobi mentioned. To be used together with change to the statestore.tf

File: templates/core/terraform/state-store/statestore.tf [hunk 2]
```diff
@@ -0,0 +1,27 @@
+resource "azurerm_cosmosdb_account" "tre-db-account" {
+  name                = var.name
```
Comment (deniscep) [inline templates/core/terraform/state-store/statestore.tf:2]:
```suggestion
  name                = "cosmos-${var.resource_name_prefix}-${var.environment}-${var.tre_id}"
```

File: templates/core/terraform/state-store/variables.tf [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+variable "location" {
+  type        = string
+  description = "location"
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "Resource group name"
+}
```
Comment (deniscep) [inline templates/core/terraform/state-store/variables.tf:14]:
```suggestion
variable "resource_name_prefix" {}
variable "environment" {}
variable "tre_id" {}
variable "location" {}
variable "resource_group_name" {}
```

File: templates/core/terraform/state-store/statestore.tf [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  }
+
+  geo_location {
+    location          = var.location
+    failover_priority = 0
+  }
+}
+
+resource "azurerm_cosmosdb_sql_database" "tre-db" {
+  name                = "tre"
```
Comment (deniscep) [inline templates/core/terraform/state-store/statestore.tf:23]:
```suggestion
  name                = "cosmos-sql-${var.resource_name_prefix}-${var.environment}-${var.tre_id}"
```

**Review by deniscep — CHANGES_REQUESTED** [review]
aligned pattern with other modules; all suggestions are interconnected.

File: templates/core/terraform/state-store/statestore.tf [hunk 5]
```diff
@@ -0,0 +1,27 @@
+resource "azurerm_cosmosdb_account" "tre-db-account" {
```
[the remark this answers is not in the document]
Comment (sachinkundu) [reply templates/core/terraform/state-store/statestore.tf:1]: Let's keep things simple for now. We can go multi region when things work and same with serverless SKU. Premature optimizations is the root of all evil.

[the remark this answers is not in the document]
Comment (marrobi) [reply templates/core/terraform/state-store/statestore.tf:1]: Can we create an issue to track this down the line. Evaluate appropriate field SKU or similar? @christoferlof

**Review by marrobi — APPROVED** [review]

**Pull request merged**